backend/src/v5/processors/users.js

```diff
@@ -72,4 +73,16 @@ Users.getAvatar = getAvatar;
 
 Users.uploadAvatar = uploadAvatar;
 
+Users.resetPasswordToken = (username) => resetPasswordToken(username);
+
+Users.resetPassword = async (username, token, newPassword) => {
+	const { customData } = await getUserByUsername(username);
+
+	if (customData?.resetPasswordToken?.token !== token || customData.resetPasswordToken.expiredAt < new Date()) {
+		throw templates.invalidToken;
+	}
+
+	await updatePassword(username, newPassword);
```

Comment:
you also need to wipe the token as you update. tokens should be single use. (can just call `$unset` in the query)

Reply:
this is happening in updatePassword model method